### DISCUSSION TITLE
Build using rush library causes incorrect behaviour of FieldArray

### DISCUSSION THREAD

**Discussion by LadaBr** [react-hook-form/react-hook-form#8583] [Q&A]
### Version Number

7.33.0

### Codesandbox/Expo snack

https://codesandbox.io/s/bold-swanson-upvg38?file=/src/App.js

### Steps to reproduce

1. Create rush project with multiple TS projects (modules)
2. Import react-hook-form from module to react app
3. Start react app
4. Edit any field and see wrong behaviour of FieldArray inputs

Dirty fields are removed from array instead of value changing to true. Array contains unwanted property "remove". Default value object is always updated to current value of input.

### Expected behaviour

FieldArray inputs should be marked as dirty correctly and default value should not change.

### What browsers are you seeing the problem on?

Chrome, Edge

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by LadaBr** [comment]
dirtyFields after input on index 0 is changed.
![image](https://user-images.githubusercontent.com/19284819/175924049-d19a5311-4347-44e9-bf93-2e5337ba605a.png)

**Reply by bluebill1049** [reply]
expected behavior mark those field not dirty, dirty fields will be marked as true

**Reply by LadaBr** [reply]
How is it expected? Array that contains random property that should not be there + index 0 magically disappearing after field value is changed. In isolated react project it works completely different. Index 0 stays and it's value changes to true.
